### ISSUE TITLE
Bug with NotificationExtenderService and OSInFocusDisplayOption.None ?

### ISSUE THREAD

**Issue by alexleutgoeb** [OneSignal/OneSignal-Android-SDK#162]
When adding a custom NotificationExtenderService like

````java
public class NotificationExtender extends NotificationExtenderService {
  @Override
  protected boolean onNotificationProcessing(OSNotificationReceivedResult receivedResult) {
    // Don't handle notification, let OneSignal framework do it
    return false;
  }
}
````

and setting

````java
OneSignal.setInFocusDisplaying(OneSignal.OSInFocusDisplayOption.None);
````

neither the NotificationOpenedHandler method, nor the NotificationReceivedHandler method is being called when the app is in the foreground.

Removing the custom service from AndroidManifest calls the NotificationOpenedHandler as expected. Is this an intended behavior or a possible bug? I'm testing this with the latest 3.4.3 release.

Thanks!

**Comment by jkasten2** [comment]
@alexleutgoeb This isn't the intended behavior. If you have setup a `NotificationExtenderService` and set `InFocusDisplaying` to `None` then the `NotificationReceivedHandler` should still fire.

We will try reproducing and look into the cause.

**Comment by alexleutgoeb** [comment]
@jkasten2 Just wanted to add that I reproduced the issue with the source project too, just uncommenting the extender service in the AndroidManifest.xml file.

After some debugging I was able to narrow down the code, the issue might be caused from here: https://github.com/OneSignal/OneSignal-Android-SDK/blob/master/OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationBundleProcessor.java#L314 

As the method returns prematurely, the `shouldDisplay()` (and therefore the OSInFocusDisplayOption) is never evaluated and there is no callback to `OneSignal.handleNotificationReceived()` anymore.

**Comment by jkasten2** [comment]
@alexleutgoeb Thanks for the details and looking into the issue. You are correct, the code needs to be adjusted here to fix the issue.

I remember at the time I was designing the extender I was trying to specifically suppress the event to make sure only one event would fire. However I can see that is can be use full sometimes to handle things from the `NotificationReceivedHandler` instead. Also so you don't have to move code around when adding an extender. So I am for changing this behavior.

Another issue is that `OSInFocusDisplayOption.None` seems to be ignored by the `NotificationExtenderService`. I will most likely change things to respect this setting from here as well after a bit more thought on deciding if this would create any issues.

**Comment by jkasten2** [comment]
@alexleutgoeb Just fixed the `NotificationReceivedHandler` bug in the commit above and it will be included in the 3.4.4 release.

Also I have decided that `OSInFocusDisplayOption.None` will not prevent the `NotificationExtenderService` from displaying a notification. This is to keep the app developer's `NotificationExtenderService` in control. `OSNotificationReceivedResult.isAppInFocus` can be read to tell if the app is in focus and the app developer can return `true` to prevent the notification from displaying if they choose.
